cluster-provision/gocli/pkg/libssh/ssh.go

```diff
@@ -16,13 +21,18 @@ var sshKey []byte
 // is going to be passed. any leading ways to configure the script like /bin/bash or anything is left to the caller to account for as an implementation detail
 type Client interface {
 	Command(cmd string) error
+	CommandWithNoStdOut(cmd string) (string, error)
+	CopyRemoteFile(remotePath, localPath string) error
```

Comment:
Could we replace the `localPath` with `io.Writer` ?

Reply:
we can
but my rationale behind this was to make this ssh client as similar as can be to the ssh commands. and these work with paths and directories and abstract the os internals. without this the responsibility of file creation is moved to the caller

Reply:
I think the flexibility is better in this case.